### ISSUE TITLE
[Dialog] When `escapeDisabled` is present keyboard navigation is restricted

### ISSUE THREAD

**Issue by geospatialem** [Esri/calcite-design-system#10363]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

When the component contains the `escape-disabled` attribute the component's focusable elements cannot be accessed, and elements cannot be modified as expected. _The behavior occurs with/without the `modal` attribute._

![demo](https://github.com/user-attachments/assets/68d01de4-f768-4664-b8cb-25dda17fe0eb)

### Expected Behavior

Similar functionality that is performed with the deprecated `modal` component.

![demo2](https://github.com/user-attachments/assets/c8b97307-faaf-4a30-9f42-baf5cb50ef0a)

### Reproduction Sample

https://codepen.io/geospatialem/pen/LYwPRdZ

### Reproduction Steps

1. Open [the sample](https://codepen.io/geospatialem/pen/LYwPRdZ)
2. Try to navigate through the `dialog` using the keyboard
3. Note that the `tab` key does not work. Try to select the `tab` and `shift` keys and observe the last button `"Save"` is tabbable, but other elements are not.
4. Try to select the `input` component with your mouse and/or change the contents of the component.

### Reproduction Version

2.12.2

### Relevant Info

Identified via an [Esri Community post](https://community.esri.com/t5/calcite-design-system-questions/calcite-dialog-when-escape-disabled/m-p/1540905#M743) on September 20, 2024.

### Regression?

No

### Priority impact

impact - p3 - not time sensitive

### Impact

Med/High - while not the default setting of the component, the experience isn't great so we should mitigate as we're able to.

### Calcite package

- [X] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [ ] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

N/A

**Comment by geospatialem** [comment]
Duplicate of https://github.com/Esri/calcite-design-system/issues/10337. Verified on the `dev` branch with the example above:

![verified](https://github.com/user-attachments/assets/dc69d4b8-909e-416c-b322-7a7b593d2e06)
